### PR TITLE
Merge deep merge rename keys

### PR DESCRIPTION
**Pull request by boxxxie**

### CODE REVIEW

**Comment by boxxxie** [comment]
@kenips

**Comment by boxxxie** [comment]
@kenips

File: README.md [hunk 1]
```diff
@@ -1 +1,24 @@
...
+
+  1. Add `extensions` to your list of dependencies in `mix.exs`:
+
+    ```elixir
+    def deps do
+      [{:kitchen_sink, "~> 0.0.1"}]
+    end
+    ```
+
+  2. Ensure `extensions` is started before your application:
```
Comment (kenips) [inline README.md]: extensions -> kitchen_sink

File: README.md [hunk 2]
```diff
@@ -1 +1,24 @@
-# pw-exex
+# Kitchen Sink
+
+Module to make improvements to the Elixir Standard API. This module should follow Elixir guidelines, and we should consider everything in these files as able to be patched into mainstream Elixir. This is a good place to adopt APIs from other languages or frameworks (Ramda, Clojure, Elm, Haskell, etc...)
+
+## Installation
+
+If [available in Hex](https://hex.pm/docs/publish), the package can be installed as:
+
+  1. Add `extensions` to your list of dependencies in `mix.exs`:
```
Comment (kenips) [inline README.md]: extensions -> kitchen_sink

File: lib/kitchensink/map.ex [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+
+  This module is a mixin for the elixir map namespace
+
+  """
+
+  import KitchenSink.Function
+  # --------------------- Map.merge/1 ----------------------------------
+  @doc """
+
+merge an array of maps! using merge/1
```
Comment (kenips) [inline lib/kitchensink/map.ex]: space

File: test/kitchensink_test/map_test.exs [hunk 4]
```diff
@@ -0,0 +1,78 @@
+defmodule KitchenSink.MapTest do
+  @moduledoc false
+
+  use ExUnit.Case
+  # import Planswell.Extensions.Map
```
Comment (robertpeacock22) [inline test/kitchensink_test/map_test.exs]: Remove if unused.

Comment (kenips) [inline test/kitchensink_test/map_test.exs]: ????

File: lib/kitchensink/map.ex [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+  rename_key remaps a value from one key to another in a map
+
+  ## Example:
+
+  iex> Map.rename_key %{a: 1, b: 2, c: 3, d: 4}, :a, :z
+  %{z: 1, b: 2, c: 3, d: 4}
+
+  """
+
+  def rename_key(map, key_map) when is_map(key_map) and map_size(map) === 1 do
```
Comment (kenips) [inline lib/kitchensink/map.ex]: i thought you'd get this for free as last argument could be keyword list.

Comment (boxxxie) [reply lib/kitchensink/map.ex]: i was unaware of this

Comment (kenips) [reply lib/kitchensink/map.ex]: you're correct. this is map not list. Perhaps we shouldn't allow map use like this.

Comment (boxxxie) [reply lib/kitchensink/map.ex]: keyword lists can contain the same keys, maybe we shouldn't use them

Comment (boxxxie) [reply lib/kitchensink/map.ex]: i may have to rewrite too much for this change... i'll do it later.

**Comment by boxxxie** [comment]
merge?

File: lib/kitchensink/map.ex (hunk 5 above)
Comment (boxxxie) [reply lib/kitchensink/map.ex]: i was wrong. keyword lists are implemented now

**Pull request merged**